### PR TITLE
[Install] Update UV installation instructions and add dedicated UV tabs

### PR DESCRIPTION
**Pull request by tonyhoo**
## Issue #, if available:*

Issue #4593

## Description of changes:
1. Added dedicated UV installation tabs for all platforms (Linux/Mac/Windows)
2. Fixed virtual environment installation issues by:
   - Using `python -m uv pip install` instead of `uv pip install`
   - Creating separate UV-specific installation files
3. Cleaned up installation files:
   - Removed UV installer from non-UV installation methods (pip, source)
   - Created dedicated install-cpu-uv.md and install-gpu-uv.md
   - Updated nightly build instructions to use correct UV command format

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/install.md [hunk 1]
```diff
@@ -27,6 +27,20 @@ The Conda install may have subtle differences in installed dependencies that cou
...
   
+  :::::{tab} UV
+  
+    ::::{tab} CPU
+    ```{include} install-cpu-uv.md
+    ```
+    ::::
+    
+    ::::{tab} GPU
+    ```{include} install-gpu-pip.md
```
Comment (Innixma) [inline docs/install.md:38]:
```suggestion
    ```{include} install-gpu-uv.md
```

File: docs/install.md [hunk 2]
```diff
@@ -130,6 +161,26 @@ The Conda install may have subtle differences in installed dependencies that cou
...
+    
+    ```{include} install-cpu-uv.md
+    ```
+    ::::
+    
+    ::::{tab} GPU
+    ```{include} install-windows-gpu.md
+    ```
+
+    ```{include} install-gpu-pip.md
```
Comment (Innixma) [inline docs/install.md:178]:
```suggestion
    ```{include} install-gpu-uv.md
```

File: docs/install-cpu-source.md [hunk 3]
```diff
@@ -1,11 +1,10 @@
 ```console
 pip install -U pip
 pip install -U setuptools wheel
-pip install -U uv
 
 # CPU version of pytorch has smaller footprint - see installation instructions in
 # pytorch documentation - https://pytorch.org/get-started/locally/
-uv pip install torch==2.4.1+cpu torchvision==0.19.1+cpu --index-url https://download.pytorch.org/whl/cpu
+pip install torch==2.4.1+cpu torchvision==0.19.1+cpu --index-url https://download.pytorch.org/whl/cpu
```
Comment (Innixma) [inline docs/install-cpu-source.md:7]: Why remove uv for cpu source install? We use uv anyways in the `full_install.sh` script. I'd recommend using UV for the torch install as well. Can also remove `pip install -U pip` and `pip install -U setuptools wheel`

```
pip install -U uv
python -m uv pip install torch==....
```

Comment (tonyhoo) [reply docs/install-cpu-source.md]: Simplified

**Review by prateekdesai04 — APPROVED** [review]
LGTM

**Comment by suzhoum** [comment]
When testing the installation on ubuntu, I got an error doing `./full_install.sh`. I think relaxing `shap>=0.44,<0.47` resolved the issue.

**Comment by tonyhoo** [comment]
> When testing the installation on ubuntu with python3.12, I got an error doing `./full_install.sh`. I think relaxing `shap>=0.44,<0.47` resolved the issue.

I will test and see if it is a common issue not captured by platform_tests, if so need to address in a separate dependency update PR before release

**Pull request merged**